src/Microsoft.Azure.WebJobs.Host/Extensions/JobHostMetadataProvider.cs

```diff
@@ -273,5 +274,14 @@ private void AddTypesFromGraph(IBindingRuleProvider root)
                 }
             }
         }
+
+        public MethodMetadata GetMethodMetadata(string methodName)
+        {
+            string disabledSettingName = string.Format(CultureInfo.InvariantCulture, "AzureWebJobs.{0}.Disabled", methodName);
```

Comment:
For implementation, I think you should be passing in a deferred getter for IFunctionIndexProvider into the JobHostMetadataProvider constructor, which will allow you to use IFunctionIndex to get function metadata that has already been computed. Note that the index has already computed IsDisabled per function, so you won't need to recompute it again as you're doing now.  That would also set us up for the future for returning additional function metadata.

Also - Need to see if there are any timing issues with when the index is created, and when we the IJobHostMetadataProvider can be called to return function metadata.

Using IFunctionIndex to get FunctionDescriptors (which contain the IsDisabled property you need) does raise again the earlier question of whether we create a new type FunctionMetadata, or use the existing FunctionDescriptor. We decided to introduce a new, nicer type, though in this method you'll basically be doing a mechanical transformation from FunctionDescriptor to FunctionMetadata.